genesis/src/main.rs

```diff
@@ -30,10 +31,14 @@ pub enum AccountFileFormat {
     Keypair,
 }
 
-fn pubkey_from_file(key_file: &str) -> Result<Pubkey, Box<dyn error::Error>> {
-    read_pubkey_file(key_file)
-        .or_else(|_| read_keypair_file(key_file).map(|keypair| keypair.pubkey()))
-        .map_err(|err| format!("Failed to read {}: {}", key_file, err).into())
+fn required_pubkey(matches: &ArgMatches<'_>, name: &str) -> Result<Pubkey, Box<dyn error::Error>> {
+    pubkey_of(matches, name).map(Ok).unwrap_or_else(|| {
```

Comment:
A little cleaner and more rusty? Enables you to remove the Err wrapper inside as well.

```suggestion
    pubkey_of(matches, name).ok().map_err(|_| {
```

Reply:
:) Even better!

Reply:
Thanks for quick review comments! I was about to add such comment for rusty way. Actually, your suggestion couldn't be used because pubkey_of is returning `Option` here.

However, based on your suggestion of API use, I've managed to clean it up a bit: 58947fac2bcf552ccb3a576be63ac7df94093920

Reply:
Haha, you're always quicker than me.... xD